src/legacy/core_plugins/telemetry/server/collectors/application_usage/telemetry_application_usage_collector.ts

```diff
@@ -0,0 +1,219 @@
+        findAll<ApplicationUsageTotal>(savedObjectsClient, { type: SAVED_OBJECTS_TOTAL_TYPE }),
+        findAll<ApplicationUsageTransactional>(savedObjectsClient, {
+          type: SAVED_OBJECTS_TRANSACTIONAL_TYPE,
+        }),
+      ]);
+
+      const applicationUsageFromTotals = rawApplicationUsageTotals.reduce(
+        (acc, { attributes: { appId, minutesOnScreen, numberOfClicks } }) => {
+          const existing = acc[appId] || { clicks_total: 0, minutes_on_screen_total: 0 };
+          return {
+            ...acc,
+            [appId]: {
+              clicks_total: numberOfClicks + existing.clicks_total,
+              clicks_30_days: 0,
+              clicks_90_days: 0,
+              minutes_on_screen_total: minutesOnScreen + existing.minutes_on_screen_total,
+              minutes_on_screen_30_days: 0,
+              minutes_on_screen_90_days: 0,
+            },
+          };
+        },
+        {} as ApplicationUsageTelemetryReport
+      );
+
+      const nowMinus30 = moment().subtract(30, 'days');
+      const nowMinus90 = moment().subtract(90, 'days');
+
+      const applicationUsage = rawApplicationUsageTransactional.reduce(
+        (acc, { attributes: { appId, minutesOnScreen, numberOfClicks, timestamp } }) => {
+          const existing = acc[appId] || {
+            clicks_total: 0,
+            clicks_30_days: 0,
+            clicks_90_days: 0,
+            minutes_on_screen_total: 0,
+            minutes_on_screen_30_days: 0,
+            minutes_on_screen_90_days: 0,
+          };
+
+          const timeOfEntry = moment(timestamp as string);
+          const isInLast30Days = timeOfEntry.isSameOrAfter(nowMinus30);
+          const isInLast90Days = timeOfEntry.isSameOrAfter(nowMinus90);
+
+          const last30Days = {
+            clicks_30_days: existing.clicks_30_days + numberOfClicks,
+            minutes_on_screen_30_days: existing.minutes_on_screen_30_days + minutesOnScreen,
+          };
+          const last90Days = {
+            clicks_90_days: existing.clicks_90_days + numberOfClicks,
+            minutes_on_screen_90_days: existing.minutes_on_screen_90_days + minutesOnScreen,
+          };
+
+          return {
+            ...acc,
+            [appId]: {
+              ...existing,
+              clicks_total: existing.clicks_total + numberOfClicks,
+              minutes_on_screen_total: existing.minutes_on_screen_total + minutesOnScreen,
+              ...(isInLast30Days ? last30Days : {}),
+              ...(isInLast90Days ? last90Days : {}),
+            },
+          };
+        },
+        applicationUsageFromTotals
+      );
+
+      return applicationUsage;
+    },
+  });
+
+  usageCollection.registerCollector(collector);
+
+  setInterval(() => rollTotals(getSavedObjectsClient()), ROLL_INDICES_INTERVAL);
+  setTimeout(() => rollTotals(getSavedObjectsClient()), ROLL_INDICES_START);
+}
+
+async function rollTotals(savedObjectsClient?: ISavedObjectsRepository) {
```

Comment:
We're using this method to aggregate data and it should _at least_ have a rudimentary test since it's also using `findAll`.